### DISCUSSION TITLE
deepmd-kit-2.2.2-cpu can't find pair style deepmd?

### DISCUSSION THREAD

**Discussion by chb-deep** [deepmodeling/deepmd-kit#2648] [Q&A]
i find a error lammps can't find pair style deepmd ? I use lmp -h, can't see it on the list，Installed with conda, the version of conda is 23.1, the version of deepmd-kit is 2.2.2, and deepmd is not found in pair styles after lmp -h input, I did execute this command in deepmd environment!

help！help！help！

**Comment by njzjz** [comment]
Thanks for reporting. I tried to reproduce it but found the style can be successfully loaded. See [this link](https://nbviewer.org/gist/njzjz/4b0a2b65efd0375dcc17ed4f89c270bc). Please provide more information to locate the problem.

**Reply by chb-deep** [reply]
conda create -n deepmd1 deepmd-kit=*=*cpu libdeepmd=*=*cpu lammps -c https://conda.deepmodeling.com -c defaults
conda activate 
(base) [shuoxing@node01 ~]$ conda activate deepmd1
(deepmd1) [shuoxing@node01 ~]$ which lmp
~/.conda/envs/deepmd1/bin/lmp
(deepmd1) [shuoxing@node01 ~]$ lmp -h

Large-scale Atomic/Molecular Massively Parallel Simulator - 23 Jun 2022 - Update 1

Usage example: /home/shuoxing/.conda/envs/deepmd1/bin/_lmp -var t 300 -echo screen -in
 in.alloy

List of command line options supported by this LAMMPS executable:

-echo none/screen/log/both  : echoing of input script (-e)
-help                       : print this help message (-h)
-in none/filename           : read input from file or stdin (default) (-i)
-kokkos on/off ...          : turn KOKKOS mode on or off (-k)
-log none/filename          : where to send log output (-l)
-mdi '<mdi flags>'          : pass flags to the MolSSI Driver Interface
-mpicolor color             : which exe in a multi-exe mpirun cmd (-m)
-cite                       : select citation reminder style (-c)
-nocite                     : disable citation reminder (-nc)
-package style ...          : invoke package command (-pk)
-partition size1 size2 ...  : assign partition sizes (-p)
-plog basename              : basename for partition logs (-pl)
-pscreen basename           : basename for partition screens (-ps)
-restart2data rfile dfile ... : convert restart to data file (-r2data)
-restart2dump rfile dgroup dstyle dfile ... 
                            : convert restart to dump file (-r2dump)
-reorder topology-specs     : processor reordering (-r)
-screen none/filename       : where to send screen output (-sc)
-skiprun                    : skip loops in run and minimize (-sr)
-suffix gpu/intel/opt/omp   : style suffix to apply (-sf)
-var varname value          : set index style variable (-v)

OS: Linux "CentOS Linux 7 (Core)" 3.10.0-1160.24.1.el7.x86_64 x86_64

Compiler: GNU C++ 7.5.0 with OpenMP 4.5
C++ standard: C++11
MPI v3.1: MPICH Version:	3.3.2
MPICH Release date:	Tue Nov 12 21:23:16 CST 2019
MPICH ABI:	13:8:1

Accelerator configuration:


Active compile time flags:

-DLAMMPS_GZIP
-DLAMMPS_PNG
-DLAMMPS_JPEG
-DLAMMPS_SMALLBIG
sizeof(smallint): 32-bit
sizeof(imageint): 32-bit
sizeof(tagint):   32-bit
sizeof(bigint):   64-bit

Available compression formats:

Extension: .gz     Command: gzip
Extension: .bz2    Command: bzip2
Extension: .xz     Command: xz
Extension: .lzma   Command: xz
Extension: .lz4    Command: lz4


Installed packages:

ASPHERE BODY CLASS2 COLLOID COLVARS CORESHELL DIPOLE EXTRA-COMPUTE EXTRA-DUMP 
EXTRA-FIX EXTRA-MOLECULE EXTRA-PAIR FEP GRANULAR KSPACE MANYBODY MC MEAM MISC 
MOLECULE OPT PERI PHONON PLUGIN PLUMED QTB REAXFF REPLICA RIGID SHOCK SRD 

List of individual style options included in this LAMMPS executable

* Atom styles:

angle           atomic          body            bond            charge          
dipole          ellipsoid       full            hybrid          line            
molecular       peri            sphere          template        tri             

* Integrate styles:

respa           verlet          verlet/split    

* Minimize styles:

cg              fire            fire/old        hftn            quickmin        
sd              

* Pair styles:

adp             agni            airebo          airebo/morse    atm             
beck            body/nparticle  body/rounded/polygon            
body/rounded/polyhedron         bop             born            born/coul/dsf   
born/coul/dsf/cs                born/coul/long  born/coul/long/cs               
born/coul/msm   born/coul/wolf  born/coul/wolf/cs               brownian        
brownian/poly   buck            buck/coul/cut   buck/coul/long  
buck/coul/long/cs               buck/coul/msm   buck/long/coul/long             
buck/mdf        colloid         comb            comb3           cosine/squared  
coul/cut        coul/cut/global coul/cut/soft   coul/debye      coul/diel       
coul/dsf        coul/exclude    coul/long       coul/long/cs    coul/long/soft  
coul/msm        coul/slater/cut coul/slater/long                coul/streitz    
coul/wolf       coul/wolf/cs    reax            dsmc            e3b             
eam             eam/alloy       eam/alloy/opt   eam/cd          eam/cd/old      
eam/fs          eam/fs/opt      eam/he          eam/opt         edip            
edip/multi      eim             extep           gauss           gauss/cut       
gayberne        gran/hertz/history              gran/hooke      
gran/hooke/history              granular        gw              gw/zbl          
harmonic/cut    hbond/dreiding/lj               hbond/dreiding/morse            
hybrid          hybrid/overlay  hybrid/scaled   lcbop           lennard/mdf     
line/lj         list            lj96/cut        lj/charmm/coul/charmm           
lj/charmm/coul/charmm/implicit  lj/charmm/coul/long             
lj/charmm/coul/long/opt         lj/charmm/coul/long/soft        
lj/charmm/coul/msm              lj/charmmfsw/coul/charmmfsh     
lj/charmmfsw/coul/long          lj/class2       lj/class2/coul/cut              
lj/class2/coul/cut/soft         lj/class2/coul/long             
lj/class2/coul/long/cs          lj/class2/coul/long/soft        lj/class2/soft  
lj/cubic        lj/cut          lj/cut/coul/cut lj/cut/coul/cut/soft            
lj/cut/coul/debye               lj/cut/coul/dsf lj/cut/coul/long                
lj/cut/coul/long/cs             lj/cut/coul/long/opt            
lj/cut/coul/long/soft           lj/cut/coul/msm lj/cut/coul/wolf                
lj/cut/dipole/cut               lj/cut/dipole/long              lj/cut/opt      
lj/cut/soft     lj/cut/tip4p/cut                lj/cut/tip4p/long               
lj/cut/tip4p/long/opt           lj/cut/tip4p/long/soft          lj/expand       
lj/expand/coul/long             lj/gromacs      lj/gromacs/coul/gromacs         
lj/long/coul/long               lj/long/coul/long/opt           
lj/long/dipole/long             lj/long/tip4p/long              lj/mdf          
lj/relres       lj/sf/dipole/sf lj/smooth       lj/smooth/linear                
lj/sf           local/density   lubricate       lubricateU      lubricateU/poly 
lubricate/poly  meam            meam/c          meam/spline     meam/sw/spline  
mie/cut         momb            morse           morse/opt       
morse/smooth/linear             morse/soft      nb3b/harmonic   nm/cut          
nm/cut/coul/cut nm/cut/coul/long                nm/cut/split    peri/eps        
peri/lps        peri/pmb        peri/ves        polymorphic     reaxff          
reax/c          rebo            resquared       soft            srp             
sw              sw/angle/table  sw/mod          table           tersoff         
tersoff/mod     tersoff/mod/c   tersoff/table   tersoff/zbl     threebody/table 
tip4p/cut       tip4p/long      tip4p/long/soft tracker         tri/lj          
ufm             ufm/opt         vashishta       vashishta/table wf/cut          
yukawa          yukawa/colloid  zbl             zero            

* Bond styles:

class2          fene            fene/expand     fene/nm         gaussian        
gromos          harmonic        harmonic/shift  harmonic/shift/cut              
hybrid          morse           nonlinear       quartic         special         
table           zero            

* Angle styles:

charmm          class2          cosine          cosine/delta    cosine/periodic 
cosine/shift    cosine/shift/exp                cosine/squared  dipole          
fourier         fourier/simple  gaussian        harmonic        hybrid          
quartic         table           zero            

* Dihedral styles:

charmm          charmmfsw       class2          cosine/shift/exp                
fourier         harmonic        helix           hybrid          multi/harmonic  
nharmonic       opls            quadratic       spherical       table           
table/cut       zero            

* Improper styles:

class2          cossq           cvff            distance        fourier         
harmonic        hybrid          ring            umbrella        zero            

* KSpace styles:

ewald           ewald/dipole    ewald/dipole/spin               ewald/disp      
ewald/disp/dipole               msm             msm/cg          pppm            
pppm/cg         pppm/dipole     pppm/dipole/spin                pppm/disp       
pppm/disp/tip4p pppm/stagger    pppm/tip4p      

* Fix styles

accelerate/cos  acks2/reax      acks2/reaxff    adapt           adapt/fep       
addforce        addtorque       append/atoms    atom/swap       ave/atom        
ave/chunk       ave/correlate   ave/correlate/long              ave/histo       
ave/histo/weight                ave/time        aveforce        balance         
bond/break      bond/create     bond/create/angle               bond/swap       
box/relax       charge/regulation               cmap            colvars         
controller      damping/cundall deform          deposit         ave/spatial     
ave/spatial/sphere              lb/pc           lb/rigid/pc/sphere              
client/md       drag            dt/reset        efield          ehex            
electron/stopping               electron/stopping/fit           enforce2d       
evaporate       external        ffl             filter/corotate flow/gauss      
freeze          gcmc            gld             gle             gravity         
grem            halt            heat            hyper/global    hyper/local     
imd             indent          ipi             langevin        lineforce       
mol/swap        momentum        momentum/chunk  move            msst            
neb             nph             nph/asphere     nph/body        nph/sphere      
nphug           npt             npt/asphere     npt/body        npt/cauchy      
npt/sphere      numdiff         numdiff/virial  nve             nve/asphere     
nve/asphere/noforce             nve/body        nve/limit       nve/line        
nve/noforce     nve/sphere      nve/tri         nvk             nvt             
nvt/asphere     nvt/body        nvt/sllod       nvt/sphere      oneway          
pafi            phonon          pimd            planeforce      plumed          
pour            press/berendsen print           property/atom   qbmsst          
qeq/comb        qeq/reaxff      qeq/reax        qtb             rattle          
reaxff/bonds    reax/c/bonds    reaxff/species  reax/c/species  recenter        
restrain        rhok            rigid           rigid/nph       rigid/nph/small 
rigid/npt       rigid/npt/small rigid/nve       rigid/nve/small rigid/nvt       
rigid/nvt/small rigid/small     setforce        shake           smd             
spring          spring/chunk    spring/rg       spring/self     srd             
store/force     store/state     temp/berendsen  temp/csld       temp/csvr       
temp/rescale    tfmc            thermal/conductivity            ti/spring       
tmd             ttm             ttm/grid        ttm/mod         tune/kspace     
vector          viscosity       viscous         viscous/sphere  
wall/body/polygon               wall/body/polyhedron            wall/colloid    
wall/ees        wall/gran       wall/gran/region                wall/harmonic   
wall/lj1043     wall/lj126      wall/lj93       wall/morse      wall/piston     
wall/reflect    wall/reflect/stochastic         wall/region     wall/region/ees 
wall/srd        widom           

* Compute styles:

ackland/atom    adf             aggregate/atom  angle           angle/local     
angmom/chunk    ave/sphere/atom basal/atom      body/local      bond            
bond/local      born/matrix     centro/atom     centroid/stress/atom            
chunk/atom      chunk/spread/atom               cluster/atom    cna/atom        
cnp/atom        com             com/chunk       contact/atom    coord/atom      
damage/atom     dihedral        dihedral/local  dilatation/atom dipole          
dipole/chunk    displace/atom   entropy/atom    erotate/asphere erotate/rigid   
erotate/sphere  erotate/sphere/atom             event/displace  fabric          
fep             fep/ta          fragment/atom   global/atom     group/group     
gyration        gyration/chunk  gyration/shape  gyration/shape/chunk            
heat/flux       hexorder/atom   hma             improper        improper/local  
inertia/chunk   ke              ke/atom         ke/rigid        momentum        
msd             msd/chunk       msd/nongauss    omega/chunk     orientorder/atom      
          
pair            pair/local      pe              pe/atom         plasticity/atom 
pressure        property/atom   property/chunk  property/local  rdf             
reduce          reduce/chunk    reduce/region   rigid/local     slice           
stress/atom     stress/cartesian                stress/cylinder 
pressure/cylinder               stress/mop      stress/mop/profile              
stress/spherical                temp            temp/asphere    temp/body       
temp/chunk      temp/com        temp/cs         temp/deform     temp/partial    
temp/profile    temp/ramp       temp/region     temp/rotate     temp/sphere     
ti              torque/chunk    vacf            vcm/chunk       viscosity/cos   

* Region styles:

block           cone            cylinder        ellipsoid       intersect       
plane           prism           sphere          union           

* Dump styles:

atom            cfg             custom          dcd             image           
local           movie           xtc             xyz             yaml            

* Command styles

balance         change_box      create_atoms    create_bonds    create_box      
delete_atoms    delete_bonds    reset_ids       kim_init        kim_interactions      
          
kim_param       kim_property    kim_query       message         server          
displace_atoms  dynamical_matrix                group2ndx       hyper           
info            minimize        ndx2group       neb             plugin          
prd             read_data       read_dump       read_restart    replicate       
rerun           reset_atom_ids  reset_mol_ids   run             set             
tad             temper          temper/grem     temper/npt      third_order     
velocity        write_coeff     write_data      write_dump      write_restart

**Reply by njzjz** [reply]
It is an expected behavior that `lmp -h` doesn't give the plugin information.

**Reply by chb-deep** [reply]
How can I fix this? To install the C++ interface, install lammps as a plugin?

**Reply by njzjz** [reply]
You can request LAMMPS developers for this feature. There is nothing we can do on the DeePMD-kit side.
